servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ProtocolCompatibilityTest.java

```diff
@@ -716,45 +717,56 @@ private CompatResponse response(final int value) {
                 }
                 return computeResponse(value);
             }
-        }).appendServiceFilter(delegate -> new Compat.CompatServiceFilter(delegate) {
-            @Override
-            public Publisher<CompatResponse> bidirectionalStreamingCall(final GrpcServiceContext ctx,
-                                                                        final Publisher<CompatRequest> req) {
-                maybeThrowFromFilter();
-                return delegate().bidirectionalStreamingCall(ctx, req);
-            }
+        };
+        final ServiceFactory serviceFactory = strategy == defaultStrategy() ? new ServiceFactory(compatService) :
+                new ServiceFactory.Builder().bidirectionalStreamingCall(strategy, compatService)
+                        .clientStreamingCall(strategy, compatService)
+                        .scalarCall(strategy, compatService)
+                        .serverStreamingCall(strategy, compatService)
+                        .build();
+
+        // FIXME(idel): remove condition around appendServiceFilter when WSS or filters execution strategy will be fixed
```

Comment:
I am not yet convinced that the failure is due to the race you mention here, can we dig a bit deeper to see why are we consistently seeing exceptions without this change in the test here?

Reply:
As we discussed offline, when the offloading is enabled, it's expected that the client may receive either grpc-exception first or see that server closed the connection. See `grpcJavaToServiceTalkErrorInStreamingResponse` as an example. 
Tests with `noOffloadsStrategy()` always expect grpc-exception. Only tests with `*NoOffload` suffix are affected by this `FIXME` and it will be removed when we fix wrapping routes with filters because currently, filters override route's execution strategy.